src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ReferralRepositoryTest.kt

```diff
@@ -1,59 +1,312 @@
+    eosR: EndOfServiceReport? = null,
+  ): Multiple {
+
+    val user = authUserFactory.create(random(6), random(5), random(12))
+    val serviceProvider = serviceProviderFactory.create(random(13), random(14))
+    val contract = when {
+      asPrime -> dynamicFrameworkContractFactory.create(primeProvider = serviceProvider, contractReference = random(10))
+      else -> dynamicFrameworkContractFactory.create(subcontractorProviders = setOf(serviceProvider), contractReference = random(10))
+    }
+    val intervention = interventionFactory.create(contract = contract)
+    val referral = referralFactory.createEnded(
+      intervention = intervention,
+      assignments = listOf(ReferralAssignment(OffsetDateTime.now(), assignedBy = user, assignedTo = user)),
+      endRequestedAt = endRequestedAt,
+      concludedAt = concludedAt,
+      endOfServiceReport = eosR
+    )
+    val serviceUser = serviceUserFactory.create(random(15), random(16), referral)
+    return Multiple(referral.currentAssignee, serviceProvider, contract, intervention, referral, serviceUser)
+  }
+
+  private fun createDraftReferral(
+    asPrime: Boolean
+  ): Multiple {
+
+    val serviceProvider = serviceProviderFactory.create(random(13), random(14))
+    val contract = when {
+      asPrime -> dynamicFrameworkContractFactory.create(primeProvider = serviceProvider, contractReference = random(10))
+      else -> dynamicFrameworkContractFactory.create(subcontractorProviders = setOf(serviceProvider), contractReference = random(10))
+    }
+    val intervention = interventionFactory.create(contract = contract)
+    val referral = referralFactory.createDraft(intervention = intervention)
+    val serviceUser = serviceUserFactory.create(random(15), random(16), referral)
+    return Multiple(referral.currentAssignee, serviceProvider, contract, intervention, referral, serviceUser)
+  }
+
+  private fun random(length: Int): String {
+    return RandomStringUtils.randomAlphabetic(length)
+  }
+
+  private fun contains(summaries: List<ServiceProviderSentReferralSummary>, summary: Summary): Boolean {
+    return summaries.any {
+      it.sentAt == summary.sentAt &&
+        it.referenceNumber == summary.referenceNumber &&
+        it.interventionTitle == summary.interventionTitle &&
+        it.dynamicFrameWorkContractId == summary.dynamicFrameWorkContractId &&
+        it.assignedToUserName == summary.assignedToUserName &&
+        it.serviceUserFirstName == summary.serviceUserFirstName &&
+        it.serviceUserLastName == summary.serviceUserLastName
+    }
+  }
+
+  private fun expectedSummary(referral: Multiple) =
+    Summary(
+      referral.referral.sentAt!!.toInstant(),
+      referral.referral.referenceNumber!!,
+      referral.intervention.title,
+      referral.contract.contractReference,
+      referral.referral.currentAssignee?.id,
+      referral.serviceUser.firstName,
+      referral.serviceUser.lastName
+    )
 }
+
+data class Multiple(
```

Comment:
I'm a bit confused about the use of this class.
Why can't you just use Referral object?

Reply:
It was just a convenience although all available via referral so will do as you say

Reply:
Done

Reply:
This has now gone